TST: add azure CI for Windows tests

tylerjereddy opened the pull request:
testing in my fork for now

ilayn committed:
DEP: Remove deprecated functions for 1.2.0

potpath committed:
DOC: linalg: fix lstsq docstring on residues shape

```python
>>> import scipy.linalg
>>> scipy.linalg.lstsq([[1]], [1])[1].shape
(0,)
```

See also: https://github.com/scipy/scipy/blob/0a9e93e23453cce2f5e861bf6dc4168d788fdb51/scipy/linalg/basic.py#L1246

andyfaff commented:
Where numpy/scipy go other projects follow. Is there a good tutorial you can recommend to set up this CI for other projects/packages?

ilayn committed:
Merge pull request #9477 from potpath/patch-1

DOC: linalg: fix lstsq docstring on residues shape

tylerjereddy commented:
Here's a place to start for Python projects: https://docs.microsoft.com/en-us/azure/devops/pipelines/languages/python?view=vsts#build-environment

The documentation seems to be slowly improving over time. Most of the directives used in their CI are actually open source, which is nice. `numba` was an early adopter of Azure so I templated off them a bit--they have a huge test matrix for Azure. The NumPy one I did is probably easier to understand initally since it is all in one file.

The web interface / github marketplace app for Azure set up can take a little effort to set up, but I think it is already easier now than before.

ilayn committed:
MAINT: linalg: lstsq clean up

souravsingh committed:
DOC: Add docstring examples for rosen functions (#9478)

* DOC: Add docstring examples for rosen functions

souravsingh committed:
DOC: Add docstring example for ai_zeros and bi_zeros (#9479)

* DOC: special: Add docstring example for ai_zeros and bi_zeros

tylerjereddy committed:
BUG: fix sparse random int handling

* TestConstructUtils::test_random_sampling fails
in some SciPy wheel build scenarios because of
incorrect handling of np.int32/64 limits in the
compatibility layer; this commit attempts to fix
that behavior

WarrenWeckesser committed:
MAINT: stats: Improve implementation of chi2 ppf method.

The old implementation, self._isf(1.0-p, df), loses precision for very
small p.  The formulation 2*special.gammaincinv(df/2, p) avoids the
loss of precision.

rgommers committed:
DOC: roadmap update for changes over the last year.

Plus some evolving insights ...

stefanv committed:
Clarify ndimage section of roadmap

tylerjereddy committed:
TST: 32-bit test_distance.py now passes

* the floating point precision stringency
of test_pdist_jensenshannon_iris_nonC() has
been adjusted to allow it to pass with
a 32-bit Python interpreter, as required
by the wheels infrastructure

tylerjereddy committed:
TST: test_kolmogorov xfail 32-bit

* test_oneovernclose() was failing in 32-bit
Python interpreter tests in our wheel build
infrastructure -- marked as xfail for now

ev-br committed:
Merge pull request #9492 from WarrenWeckesser/chi2-ppf

MAINT: stats: Improve implementation of chi2 ppf method.

pvanmulbregt committed:
DOC: Replace Scipy with SciPy in the rst doc files for consistency.

Many rst files use the terms "SciPy" in one part of the document and "Scipy"
in a later. Change all "Scipy" to "SciPy".

ilayn committed:
CI: Use released NumPy and newer Cython [ci skip]

JakobStruye committed:
MAINT: scipy.sparse: Fixed error when reshaping to illegal shape (#9472)

ilayn committed:
Merge pull request #9499 from pvanmulbregt/SciPyCaps

DOC: Replace "Scipy" with "SciPy" in the .rst doc files for consistency.

ilayn committed:
DOC: Add the correct terminal type for MSYS2 [ci skip]

pvanmulbregt committed:
DOC: Replace "Scipy" with "SciPy" in .py and .txt doc files for consistency.

Explicitly excluded are the old release notes: doc/release/*-notes.rst.

rgommers committed:
Merge pull request #9504 from pvanmulbregt/SciPyCaps2

DOC: Replace "Scipy" with "SciPy" in .py and .txt doc files for consistency

rgommers committed:
Merge pull request #9493 from tylerjereddy/jensenshannon_eps_fix

TST: 32-bit test_distance.py now passes

rgommers committed:
Merge pull request #9503 from ilayn/winbuild

DOC: Add the correct terminal type for MSYS2

ilayn committed:
Merge pull request #9502 from ilayn/pypy3_update

CI: Use released NumPy and newer Cython

rgommers committed:
DOC: address review comments, add note on linprog to roadmap.

ilayn committed:
DEP: Remove deprecated functions from interpolate

Xref #1408

ilayn committed:
CI: MAINT: Skip ckdtree test on pypy [ci skip]

ilayn committed:
CI: MAINT: skip test_arnoldi on pypy runs

rgommers committed:
Merge pull request #9494 from tylerjereddy/test_kolmogorov_32bit

TST: test_kolmogorov xfail 32-bit

rgommers committed:
Merge pull request #9325 from ilayn/rm_misc_funcs

DEP: Remove deprecated functions for 1.3.0

ilayn committed:
Merge pull request #9507 from ilayn/pypy_skip_test

CI: MAINT: Skip a ckdtree test on pypy

ilayn committed:
DOC: Update rel notes for removed funcs/aliases [ci skip]

ev-br committed:
DEP: interpolate: remove deprecated interpolate_wrapper

Removed for Scipy 1.3.0, has been deprecated since 1.0.0.

rgommers committed:
Merge pull request #9486 from tylerjereddy/fix_sparse_int_compat

BUG: fix sparse random int handling

ilayn committed:
DOC: Add deprecated alias functions [ci skip]

rgommers committed:
Merge pull request #9510 from ilayn/rel_notes_rm

DOC: Update 1.3.0 release notes for removed misc and interpolate funcs/aliases

rgommers committed:
Merge pull request #9511 from ev-br/rm_interpolate_wrapper

DEP: interpolate: remove deprecated interpolate_wrapper

pvanmulbregt committed:
DOC: Replace "Numpy" with "NumPy" for consistency in the documentation:

Change Numpy in .rst and .txt doc files, and in the doc of .py files.
Release notes untouched.

pvanmulbregt committed:
DOC: Replace "Numpy" with "NumPy" for consistency in the doc (2 more files)

tylerjereddy committed:
TST: test_random_sampling 32-bit handling

* test_random_sampling now marked as xfail
for NumPy versions prior to 1.11.0 when used
in a 32-bit Python interpreter, based on observed
wheel build failures

figiel committed:
BUG: improve error handling in stats.iqr

Add missing check for NaNs in rng which is expected by TestIQR.test_rng
test.

Closes #9516.

rgommers committed:
Merge pull request #9517 from figiel/gh-9516

BUG: improve error handling in stats.iqr

rgommers committed:
Merge pull request #9512 from tylerjereddy/xfail_random_sampling

TST: test_random_sampling 32-bit handling

rgommers committed:
Merge pull request #9480 from ilayn/lstsq_clnup

MAINT: linalg: lstsq clean up

figiel committed:
TST: relax precision requirements in signal.correlate tests

Correlate when used with the FFT method doesn't benefit from 128-bit
float types because of FFT implementation down casts input arrays to
doubles.
For this reason reduce the expected precision to the same amount as
it's expected for the double type.

Closes #9520.

rgommers committed:
DOC: fix missing random seed in optimize.newton example

This was giving occasional CI failures in the refguide check that looked like:

```
scipy.optimize.newton
---------------------
/home/travis/build/scipy/scipy/build/testenv/lib/python3.6/site-packages/scipy/optimize/zeros.py:439: RuntimeWarning: some failed to converge after 50 iterations
  warnings.warn(msg, RuntimeWarning)
File "build/testenv/lib/python3.6/site-packages/scipy/optimize/zeros.py", line 242, in newton
Failed example:
    np.allclose(vec_res, loop_res)
Expected:
    True
Got:
    False
```

ev-br committed:
Merge pull request #9529 from rgommers/fix-newton-example

DOC: fix missing random seed in optimize.newton example

rgommers committed:
Merge pull request #9526 from figiel/gh-9520

TST: relax precision requirements in signal.correlate tests

rgommers committed:
Merge pull request #9489 from rgommers/roadmap-update

DOC: roadmap update for changes over the last year.

rgommers committed:
Merge pull request #9538 from pvanmulbregt/NumPyCaps

DOC: Replace "Numpy" with "NumPy" in .py, .rst and .txt doc files for consistency

tylerjereddy committed:
TST: add Windows Azure CI
